pkg/middleware/middleware.go

```diff
@@ -67,6 +70,15 @@ func GetContextHandler(ats m.UserTokenService, remoteCache *remotecache.RemoteCa
 				ctx.Logger.Error("Failed to update last_seen_at", "error", err)
 			}
 		}
+
```

Comment:
I tried to setup a transaction in the middleware. The reasoning in my mind is that by default you probably want any API request to be in transaction and this would make it easier to not forget that and there will be less context passing around on the other hand it is more implicit.

Reply:
I have thought about doing something like this as well. It's a very common approach coming from the .Net world for example. However, for this to work I think you'll need to use the context from the `*macaron.Context` - there should be a `context.Context` exposed there I think.

Reply:
Can this be in a separate middleware executing before this middleware since the `initContextWith...` also needs transaction in some cases?

Reply:
In other comments others are not keen on creating the transaction implicitly, which at the end I agree (cases where you do not need transaction at all or want to start/end after/before the request start/end) but if you think this is a good approach I would like to hear your opinion.